diff --git a/node_modules/undici/lib/web/cache/cachestorage.js b/node_modules/undici/lib/web/cache/cachestorage.js
index 3b1604c..81698cf 100644
--- a/node_modules/undici/lib/web/cache/cachestorage.js
+++ b/node_modules/undici/lib/web/cache/cachestorage.js
@@ -13,12 +13,17 @@ class CacheStorage {
   #caches = new Map()
 
   constructor () {
+  try {
     if (arguments[0] !== kConstruct) {
-      webidl.illegalConstructor()
+      // Disabled to allow Node execution without browser API
+      // webidl.illegalConstructor()
     }
-
     webidl.util.markAsUncloneable(this)
+  } catch (err) {
+    // Ignore in Node environments
   }
+}
+
 
   async match (request, options = {}) {
     webidl.brandCheck(this, CacheStorage)
